### ISSUE TITLE
Document Cleanup

### ISSUE THREAD

**Issue by tkellen** [json-api/json-api#659]
Before we mark 1.0, we need to review the specification and do some cleanup to address the now scattered nature of the document. Below, a checklist of things we want to review:
- [x] clean up definition of meta and semantics around use
- [x] clean up definition of links and the semantics around use
- [x] replace references to URLs with the more general "links" (which can be a string or link object)
- [x] clean up bulleted lists to use the same style of prose everywhere
- [x] review use of "" and *\* around keywords/links
- [x] clean up usages of `-` and `:` after member names
- [x] review use of quoted vs. unquoted member names (e.g. `id` vs. `"id"`)
- [x] remove Content-Type from examples without content (e.g. GET, DELETE)

**Comment by cziegenberg** [comment]
I just created PR #663, to avoid conflicts with future versions and the implementation of deferred features (like the extensions), This should be merged before 1.0.

**Comment by tkellen** [comment]
boom.

**Comment by dgeb** [comment]
:tada:
